internal/replacements/replacements.go

```diff
@@ -246,6 +247,32 @@ func processGroupNeedsRemovalForPod(cluster *fdbv1beta2.FoundationDBCluster, pod
 		}
 	}
 
+	desiredPod, err := internal.GetPod(cluster, processGroupStatus)
+	if err != nil {
+		return false, err
+	}
+	// Some k8s instances have security context vetting which may edit the spec automatically.
+	// This would cause changes to security context on a pod or container
+	// to constantly be seen as having a security context change, hence we want to feature guard this
+	// and also guard on the spec hash below
+	// https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/
+	if replaceOnSecurityContextChange {
+		spec, err := internal.GetPodSpec(cluster, processGroupStatus)
+		if err != nil {
+			return false, err
+		}
+		specHash, err := internal.GetPodSpecHash(cluster, processGroupStatus, spec)
+		if err != nil {
+			return false, err
+		}
```

Comment:
Do you think it makes sense to get the `spec` and `specHash` once in this method instead of fetching the same information in multiple if branches above?

Reply:
I was basically trying to avoid grabbing it for settings that don't need the check, but I can definitely do that as it would be much cleaner (could avoid with if (any of the settings), but that's also ehh)